### PR TITLE
unify parameters for buildings addons

### PR DESCRIPTION
**Pull request by juleshaas**
MR unifys parameters for the buildings addons (`m.analyse.buildings`), especially it:

* labels parameters as required
* uses parser standard options
* renames parameters, options etc. to English
* adapts/unifys label text
* reorders some parameters
* sets default values for required parameters
  * if output is previously set to default, then this value is set as input in next addon -> note: if output is named differently by the user, then default input must also be adjusted for the next module

### CODE REVIEW

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py [hunk 1]
```diff
@@ -71,36 +66,37 @@
...
 
 # %option
-# % key: ndvi_thresh
+# % key: ndvi_perc
 # % type: integer
 # % required: no
 # % multiple: no
-# % label: Define fix NDVI threshold (on a scale from 0-255) instead of estimating it from FNK
+# % label: NDVI percentile in vegetated areas to use for thresholding
+# TODO: make radio button and add default
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py:82]: radio button missing? (and afterwards remove todo)

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py [hunk 2]
```diff
@@ -71,36 +66,37 @@
...
 
 # %option
-# % key: ndvi_perc
+# % key: ndvi_thresh
 # % type: integer
 # % required: no
 # % multiple: no
-# % label: NDVI percentile in vegetated areas to use for thresholding
+# % label: Fix NDVI threshold (on a scale from 0-255) instead of estimated value from ndvi_perc and FNK
+# TODO: make radio button and add default
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py:93]: s.o.

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py [hunk 3]
```diff
@@ -71,36 +66,37 @@
...
+# # % answer: 145
 # % guisection: Parameters
 # %end
 
 # %option G_OPT_V_OUTPUT
-# % key: output
-# % type: string
-# % required: yes
-# % label: Name for output vector map
+# % answer: buildings_Bottrop_2017
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py:99]: shouldn't it be `buildings` or somehting like this. Not region specific

File: grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py [hunk 4]
```diff
@@ -28,93 +28,87 @@
...
 # %option
-# % key: gb_thresh
+# % key: gb_perc
 # % type: integer
 # % required: no
 # % multiple: no
-# % label: Define fix Green-Blue-Ratio threshold (on a scale from 0-255)
+# % label: Green-Blue-Ratio percentile in green areas to use for thresholding
+
+# TODO: make radio button and add default
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py:99]: radio button? and remove todo

File: grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py [hunk 5]
```diff
@@ -28,93 +28,87 @@
...
 
 # %option
-# % key: gb_perc
+# % key: gb_thresh
 # % type: integer
 # % required: no
 # % multiple: no
-# % label: Define Green-Blue-Ratio threshold as this percentile of green areas
+# % label: Fix Green-Blue-Ratio threshold (on a scale from 0-255)
+# TODO: make radio button and add default
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py:110]: s.o.

File: grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py [hunk 6]
```diff
@@ -28,19 +28,21 @@
 
 # %option G_OPT_V_INPUT
 # %label: Name of the input vector layer
+# % answer: buildings_Bottrop_2017
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py:31]: s.o.

File: grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py [hunk 7]
```diff
@@ -50,14 +52,15 @@
...
-# % required: no
+# % required: yes
 # % multiple: no
 # % label: Maximum value of fractal dimension of identified change areas (see v.to.db)
 # % answer: 2.5
 # % guisection: Parameters
 # %end
 
 # %option G_OPT_V_OUTPUT
+# % answer: buildings_alkis_diff_Bottrop_2017
```
Comment (linakrisztian) [inline grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py:63]: maybe no default here, cause region specific? (see also import)

**Review by linakrisztian — COMMENTED** [review]
I have some remarks, but overall looks good to me, thanks:)

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py (hunk 1 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py:82]: I would do this in a separate MR for various addons.

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py (hunk 2 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py:93]: I would do this in a separate MR for various addons.

File: grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py (hunk 4 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py:99]: I would do this in a separate MR for various addons.

File: grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py (hunk 5 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/r.extract.greenroofs/r.extract.greenroofs.py:110]: I would do this in a separate MR for various addons.

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py (hunk 3 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py:99]: Good question. It was wished for default values according to the documentation. However the example is from `r.trees.peak` and `r.tree.mlapply`, there the default values are more generic like `tree_peaks`.

We could either do the defaults like in the documentation (also region specific) or adjust the examples in the documentation as well and use non-region-specific defaults.

File: grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py (hunk 6 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py:31]: see above

File: grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py (hunk 7 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/v.cd.areas/v.cd.areas.py:63]: see above

File: grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py (hunk 3 above)
Comment (juleshaas) [reply grass-gis-addons/m.analyse.buildings/r.extract.buildings/r.extract.buildings.py]: Yes, I agree. I changed the defaults to something more generic. A specific example can be added to the documentation.

**Pull request merged**